python/pyarrow/compute.py

```diff
@@ -180,12 +180,22 @@ def _handle_options(name, option_class, options, kwargs):
     return options
 
 
-def _make_generic_wrapper(func_name, func, option_class):
+def _make_generic_wrapper(func_name, func, option_class, arity):
     if option_class is None:
         def wrapper(*args, memory_pool=None):
+            if arity is not None and len(args) != arity:
+                raise TypeError(
+                    "{} takes {} positional argument(s), but {} were "
+                    "given".format(func_name, arity, len(args))
```

Comment:
Nit: we can use format strings nowadays :-)

Reply:
Ah, I actually wrote them first that way and then converted to this thinking we can't use it yet .. Will convert back :)